components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -2271,21 +2263,42 @@ public static boolean isStepBasedSequenceHandlerExecuted(AuthenticationContext c
      */
     public static String[] getMissingClaims(AuthenticationContext context) {
 
-        StringBuilder missingClaimsString = new StringBuilder();
-        StringBuilder missingClaimValuesString = new StringBuilder();
+        StringJoiner missingClaimsString = new StringJoiner(",");
+        StringJoiner missingClaimValuesString = new StringJoiner(",");
 
         Map<String, String> missingClaims = getMissingClaimsMap(context);
 
         for (Map.Entry<String, String> entry : missingClaims.entrySet()) {
-            missingClaimsString.append(entry.getKey());
-            missingClaimValuesString.append(entry.getValue());
-            missingClaimsString.append(",");
-            missingClaimValuesString.append(",");
+            missingClaimsString.add(entry.getKey());
+            missingClaimValuesString.add(entry.getValue());
         }
 
         return new String[]{missingClaimsString.toString(), missingClaimValuesString.toString()};
     }
 
+    /**
+     * To get display names of missing mandatory claims from SP side.
+     *
+     * @param missingClaimMap mandatory claim's URIs.
+     * @param localClaims all claims
+     * @return set of display names of missing claims
+     */
+
+    public static String[] getMissingClaimsDisplayNames(Map<String, String> missingClaimMap, List<LocalClaim> localClaims) {
+
+        StringJoiner displayNameMappingString = new StringJoiner(",");
+        for (Map.Entry<String, String> entry : missingClaimMap.entrySet()) {
+            for (LocalClaim localClaim : localClaims){
+                if (entry.getValue().equalsIgnoreCase(localClaim.getClaimURI())) {
+                    displayNameMappingString.add(entry.getKey() +"|"+localClaim.getClaimProperties().get(DISPLAY_NAME_PROPERTY));
+                    break;
+                }
+            }
+        }
+
+        return new String[]{displayNameMappingString.toString()};
```

Comment:
Since this is actually only returning one formatted string there is no need to have a return type of an array.